candlepin/candlepin
# 1412113: ensure one sub pool per stack post batch bind

This PR undoes #1450 with a much simpler fix that I missed the first time. It addresses all the feedback on the first PR, except:
* Once looking up sub pools given a set of stack ids is CandlepinQuer-ized, we should take advantage of that here
* There is a lot of method jumping in post bind operations, we need some refactoring there.

To test the PR:

Verify test failed earlier:
1. git checkout candlepin-2.0.23-1
2. deploy candlepin
3. git checkout vritant/1450
4. buildr rspec:autobind_spec.rb

Verify test passes now:
1. git checkout vritant/1450
2. deploy candlepin
3. buildr rspec:autobind_spec.rb
